### PR TITLE
feat: return supported protocols in id output

### PR DESCRIPTION
**Pull request by Stebalien**
This is really useful for debugging.

TODO: fix tests if @aschmahmann and @achingbrain OK this change.

### CODE REVIEW

**Comment by jacobheun** [comment]
I was trying to do this the other day, this would be great to have support for.

File: core/commands/id.go [hunk 1]
```diff
@@ -97,15 +99,17 @@ EXAMPLE:
 			return errors.New(offlineIdErrorMessage)
 		}
 
-		p, err := n.Routing.FindPeer(req.Context, id)
-		if err == kb.ErrLookupFailure {
+		// We need to actually connect to run identify.
+		err = n.PeerHost.Connect(req.Context, peer.AddrInfo{ID: id})
```
Comment (aschmahmann) [inline core/commands/id.go:103]: This is a fine conceptual change. Just a note that this doesn't actually do anything as the DHT (our only PeerRouting system) will connect to the peer, and will not return any addresses if it fails. https://github.com/libp2p/go-libp2p-kad-dht/blob/fc3558cc35274f5d1727dad8201f821df1ee6317/routing.go#L675-L682

File: core/commands/id.go [hunk 2]
```diff
@@ -163,6 +168,13 @@ func printPeer(ps pstore.Peerstore, p peer.ID) (interface{}, error) {
 	for _, a := range addrs {
 		info.Addresses = append(info.Addresses, a.String())
 	}
+	sort.Strings(info.Addresses)
```
Comment (aschmahmann) [inline core/commands/id.go:171]: I added address sorting to this function, any objections?

File: core/commands/id.go [hunk 3]
```diff
@@ -198,6 +210,9 @@ func printSelf(node *core.IpfsNode) (interface{}, error) {
 		for _, a := range addrs {
 			info.Addresses = append(info.Addresses, a.String())
 		}
+		sort.Strings(info.Addresses)
```
Comment (aschmahmann) [inline core/commands/id.go:213]: I added address sorting to this function, any objections?

**Comment by jacobheun** [comment]
> are we ok merging this change without the corresponding js-ipfs one, or do we need to block on that?

No need to block on JS, I created https://github.com/ipfs/js-ipfs/issues/3219 to track adding support there.

File: core/commands/id.go (hunk 1 above)
Comment (Stebalien) [reply core/commands/id.go:103]: Hm. I thought I had some trouble where we weren't actually _waiting_ for the connect event to finish.

But honestly, we should do this anyways. We shouldn't assume that the router will always end up connecting to the target peer.

File: core/commands/id.go (hunk 2 above)
Comment (Stebalien) [reply core/commands/id.go:171]: LGTM!

**Comment by achingbrain** [comment]
This is being added to js in https://github.com/ipfs/js-ipfs/pull/3250

**Pull request merged**